eZ/Publish/Core/MVC/Symfony/SiteAccess/Matcher/HostElement.php

```diff
@@ -37,10 +37,15 @@ class HostElement implements VersatileMatcher
     /**
      * Constructor.
      *
-     * @param int $elementNumber Number of elements to take into account.
+     * @param int|array $elementNumber Number of elements to take into account.
      */
     public function __construct($elementNumber)
     {
+        if (is_array($elementNumber) && isset($elementNumber['value'])) {
```

Comment:
To iterate on what I said in the other PR, and made more clear now with the link to Config class. I think I would prefer if we clean this up, and streamline it with the other matchers, as it obviously seems only arrays are supported here from config injection side.

Meaning:
```php
     * @param array $siteAccessesConfiguration SiteAccesses configuration hash, with elementNumber on `value` key.
     */
    public function __construct(array $siteAccessesConfiguration)
    {
        $this->elementNumber = isset($siteAccessesConfiguration['value']) ? (int)$siteAccessesConfiguration['value'] : 0;
```

And adjusting tests for this, wdt @bdunogier @lolautruche.